### PR TITLE
Add public API to build provider-specific convention sets without using DI

### LINKED ISSUE #3529: Make it easy to create a provider configured ModelBuilder
One scenario we want to support is external model creation via ModelBuilder. However, it is currently difficult to get a ModelBuilder instance that has been configured with a provider specific convention set.

cc @sebastienros

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #3529

### CODE REVIEW

File: test/EntityFramework.Sqlite.Tests/Metadata/Conventions/SqliteConventionSetBuilderTests.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        public override IModel Can_build_a_model_with_default_conventions_without_DI()
+        {
+            var model = base.Can_build_a_model_with_default_conventions_without_DI();
+
+            Assert.Equal("Product", model.GetEntityTypes().Single().Sqlite().TableName);
+
+            return model;
+        }
+
+        protected override ConventionSet GetConventionSet() => base.GetConventionSet();
```
Comment (smitpatel) [inline test/EntityFramework.Sqlite.Tests/Metadata/Conventions/SqliteConventionSetBuilderTests.cs]: `SqliteConventionSetBuilder.Build()` ?

File: test/EntityFramework.MicrosoftSqlServer.Tests/Metadata/Conventions/SqlServerConventionSetBuilderTests.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Microsoft.Data.Entity.SqlServer.Metadata.Conventions.Tests
+{
+    public class SqlServerConventionSetBuilderTests : ConventionSetBuilderTests
+    {
+        public override IModel Can_build_a_model_with_default_conventions_without_DI()
+        {
+            var model = base.Can_build_a_model_with_default_conventions_without_DI();
+
+            Assert.Equal("Product", model.GetEntityTypes().Single().SqlServer().TableName);
```
Comment (smitpatel) [inline test/EntityFramework.MicrosoftSqlServer.Tests/Metadata/Conventions/SqlServerConventionSetBuilderTests.cs]: What does this assert actually verify? Isn't table name is going to be `Product` in SqlServer/SQLite/Core all?

File: test/EntityFramework.Sqlite.Tests/Metadata/Conventions/SqliteConventionSetBuilderTests.cs (hunk 1 above)
Comment (AndriySvyryd) [inline test/EntityFramework.Sqlite.Tests/Metadata/Conventions/SqliteConventionSetBuilderTests.cs]: Yes... Hmmm this test is useless, will change

**Comment by smitpatel** [comment]
:shipit:

**Pull request merged**